src/ucp/wireup/wireup_cm.c

```diff
@@ -123,7 +209,10 @@ static ssize_t ucp_cm_client_priv_pack_cb(void *arg,
     ucp_lane_index_t lane_idx;
     ucp_rsc_index_t rsc_idx;
     const char *dev_name;
-    ucp_ep_h tmp_ep;
+    ucp_ep_h tmp_ep_old, tmp_ep_new;
```

Comment:
in case this function was already called for the first cm in the list, tmp_ep already exists (tmp_ep_old). so for the next cm, for which this function is called, we create tmp_ep_new.
we always exit this function with a single tmp_ep. old/new are just used to ease the pending ops reassigning.

Reply:
so there is no any moment when 2 are exist at the same time, right?

Reply:
we create tmp_ep_new in any way. if tmp_ep_old already exists (second call to client_priv_pack_cb), then we clean it up, and reset cm_wireup_ep->tmp_ep = tmp_ep_new;
i can move the cleanup of tmp_ep_old to before the creation of tmp_ep_new.

Reply:
my comment is about local vars, if single tmp var is enough, then better to reuse that var since it's temporary by its name

Reply:
@evgeny-leksikov tmp_ep_new is created in any way, since we work on it before doing the assignment to cm_wireup_ep->tmp_ep.
but cm_wireup_ep->tmp_ep may already exist from the previous call to this function. i added tmp_ep_old so that i dont have to write cm_wireup_ep->tmp_ep every time when doing the cleanup of the old tmp_ep.

Reply:
no need to write `cm_wireup_ep->tmp_ep` every time, reuse `tmp_ep`